api/authenticator.py

```diff
@@ -1237,6 +1237,12 @@ def _restriction_matches(cls, field, restriction, match_type):
 
     def enforce_library_identifier_restriction(self, identifier, patrondata):
         """Does the given patron match the configured library identifier restriction?"""
+        if isinstance(patrondata, Patron):
+            if self.library_id == patrondata.library_id:
+                return patrondata
+            else:
+                return False
+
```

Comment:
Since remote_patron_lookup can take a Patron object, this method now needs to be able to handle being given a Patron object.